src/coreclr/vm/threads.cpp

```diff
@@ -1176,11 +1176,6 @@ void InitThreadManager()
     // to update the contracts if you remove this flag.
     g_DeadlockAwareCrst.Init(CrstDeadlockDetection, CRST_UNSAFE_ANYMODE);
 
-#ifdef _DEBUG
```

Comment:
I think this should stay and be initialized to `GetRandomInt(10)`. The OBJECTREF debug-only checks are not fully deterministic. The randomization allows us to find the given problem with some probability. If we delete this randomization, we will find some problems all the time and some problems never.